crates/sozo/src/ops/execute.rs

```diff
@@ -1,21 +1,69 @@
 use anyhow::{Context, Result};
 use dojo_world::metadata::Environment;
 use starknet::accounts::{Account, Call};
-use starknet::core::utils::get_selector_from_name;
+use starknet::core::types::{BlockId, BlockTag, FieldElement, FunctionCall};
+use starknet::core::utils::{
+    cairo_short_string_to_felt, get_contract_address, get_selector_from_name,
+};
+use starknet::macros::selector;
+use starknet::providers::Provider;
+use starknet_crypto::poseidon_hash_many;
 
 use crate::commands::execute::ExecuteArgs;
 
 pub async fn execute(args: ExecuteArgs, env_metadata: Option<Environment>) -> Result<()> {
-    let ExecuteArgs { contract, entrypoint, calldata, starknet, account } = args;
+    let ExecuteArgs { contract_name, entrypoint, calldata, starknet, account } = args;
 
     let provider = starknet.provider(env_metadata.as_ref())?;
 
+    let world_address = match env_metadata.as_ref() {
+        Some(env) => match env.world_address {
+            Some(ref address) => Some(address.clone()),
+            None => None,
+        },
+        None => return Err(anyhow::anyhow!("No World Address found")),
+    };
```

Comment:
We can simplify to something like:

```suggestion
let world_address = env_metadata.as_ref()
    .and_then(|env| env.world_address.as_ref())
    .map(|address| address.clone())
    .ok_or_else(|| anyhow::anyhow!("No World Address found"))?;
```